en/micro-integrator/docs/references/connectors/redis-connector/redis-connector-example.md

```diff
@@ -0,0 +1,419 @@
+
+This connector enables developers to use an external Redis server as a cache or a database in the WSO2 EI mediation logic.
+
+## What you'll build
+
+Given below is a sample scenario that demonstrates how to work with the WSO2 EI Redis Connector and access the Redis server, using Redis commands.
+
+The user sends the request for invoke to an API to get stock details. this REST call will get converted into a SOAP message and sent to the back-end service. While the response from the backend service is converted back to json and sent back to the API caller, WSO2 EI will extract stock volume details from the response, store it into configured Redis server.
+When users need to retrieve stock volumes collected, they can invoke `getstockvolumedetails` resource. This example also demonstrates how users can manipulate this stock volume details by removing unwanted items from the Redis server.
+
+> **Note**: In this scenario we need to set up the Redis Server in your local machine. Please refer the [Setting up the Redis Connector](redis-connector-configuration.md) documentation. Follow the steps listed under `Setting up the Redis server` section for setup the Redis server and `Set up the back-end service` section for setup the Stockquote service. 
+
+This example demonstrates how to use Redis connector to:
+
+1. Retrieve stock volume details from the Stockquote back-end service. While extracting the stock volume, creating a Redis hashe map and add stock volume details to the Redis server. (In this example a Redis hashes is used to store different companies' stocks volume details. Since the “symbol” that is sent in the request is “WSO2”, the request is routed to the WSO2 endpoint. Once the response from the WSO2 endpoint is received, it is transformed according to the specified template and sent to the client. Then create a hashes map and insert extracted details to the Redis hashe map).
+2. Retrieve all stock volume details from the redis server.
+4. Remove stock volume details.
+
+All four operations are exposed via an `StockQuoteAPI` API. The API with the context `/stockquote` has four resources  
+
+* `/getstockquote/{symbol}` : Used to get stock volume details. While extracting and send details in to the Redis hash map.
+* `/getstockvolumedetails` : Retrieve information about the inserted stock volume details.
+* `/deletestockvolumedetails` : Remove unwanted stock volume details.
+
+The following diagram shows the overall solution. The user creates a hash map, stores some WSO2 stock volume details into the list, and then receives it back and remove unwanted hash map items. To invoke each operation, the user uses the same API.
```

Comment:
The user creates a Redis hashmap.

Reply:
To invoke each operation, the user uses the same API - i believe this was mentioned before on the document as well.

Reply:
Updated.